### PR TITLE
Discovery article

### PR DESCRIPTION
**Pull request by coszio**
to do:
- [x] article images
- [x] preview/header image
- [x] Update publication date before merging

### CODE REVIEW

File: qdrant-landing/content/articles/discovery-search.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  - specialty
+  - search
+  - discovery
+  - state of the art
+  - vector search
+---
+
+When Christopher Columbus and his crew sailed to cross the Atlantic Ocean, they were not looking for America. They were looking for a new route to India, and they were convinced that the Earth was round. They didn't know anything about America, but since they were going west, they stumbled upon it.
+
+They couldn't reach their target, because the context of the geography didn't let them, but once they realized it wasn't India, they claimed it a new "discovery" for their crown. Let's keep this concepts of target and context in mind, as we explore the new functionality of Qdrant: __Discovery search__.
```
Comment (agourlay) [inline qdrant-landing/content/articles/discovery-search.md:24]: > They couldn’t reach their target, because the context of the geography didn’t let them.

I am not sure this analogy helps me understand the concept of `context` here.
What is the context of geography exactly?

File: qdrant-landing/content/articles/discovery-search.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+One previous way to restrict the points that are available to the search is by filtering out the points we know for a fact are not relevant. This is really powerful because it allows us to craft complex filters that show only the points that satisfy their criteria deterministically. However, the data –or payload– associated to each point is arbitrary and does not tell us anything about their position in the vector space. In other words, it would create a _mask_ rather than a _boundary_ in the space.
+
+This is where the concept of __vector _context___ comes in. We define _context_ as a list of pairs, which in turn are made up of a positive and a negative vector. By adding a context, we can define boundaries within the vector space, to which the positive side will always be preferred over the negative side, effectively partitioning the space in which the search is performed. After the space is partitioned, we then need a _target_ to select the points that are closest to it.
```
Comment (agourlay) [inline qdrant-landing/content/articles/discovery-search.md:30]: The word `boundary` is a bit fuzzy for higher dimension space.
Is it like a hyperplane cutting orthogonal between the positive and negative vector?

Comment (coszio) [reply qdrant-landing/content/articles/discovery-search.md:30]: Exactly! I like the way you are saying it

File: qdrant-landing/content/articles/discovery-search.md (hunk 1 above)
Comment (coszio) [reply qdrant-landing/content/articles/discovery-search.md:24]: The idea is to give a notion as to why sometimes it is important for a search to be tied to a context. I see now that saying "context of geography" is ambiguous. Let me rephrase.

File: qdrant-landing/content/articles/discovery-search.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- Can't see it again
+- Cannot do reverse image search.
+
+A first strategy would be describing it with text, and then looking for similar images.
+
+![Googling dragon with human heads](/articles_data/discovery-search/googling-weird-dragon.png)
+
+Trial and error might work if you are clever enough to describe it well and get close enough on the selected image. If not, good luck getting out of the similarity bubble, even if you can select multiple images. Why is this hard?
+
+Turns out, cross modal encoders might not work how you expect them to. Images and text are embedded in the same space, but they are not necessarily close to each other. This means that we can create a mental model of the distribution as two separate planes, one for images and one for text.
```
Comment (kacperlukawski) [inline qdrant-landing/content/articles/discovery-search.md:61]:
```suggestion
Turns out, multimodal encoders might not work how you expect them to. Images and text are embedded in the same space, but they are not necessarily close to each other. This means that we can create a mental model of the distribution as two separate planes, one for images and one for text.
```

There are also cross-encoders so I thought it might be confusing.

Comment (kacperlukawski) [inline qdrant-landing/content/articles/discovery-search.md:61]: Putting a link as a reference: https://twitter.com/metasemantic/status/1356406256802607112. There is also a pretty interesting visual.

File: qdrant-landing/content/articles/discovery-search.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+## Context search
+
+Now, second case: only context.
+
+Ever been caught in the same recommendations on your favourite music streaming service? This may be caused by getting stuck in a similarity bubble. As user input gets more complex, diversity becomes scarse, and it becomes harder to force the system to recommend something different.
+
+![Context vs recommendation search](/articles_data/discovery-search/context-vs-recommendation.png)
+
+__Context search__ solves this by de-focusing the search around a single point. Instead, it selects points randomly from within an zone in the vector space. This search is the most influenced by _triplet loss_, as the score can be thought of as _"how much a point is closer to a negative than a positive vector?"_. If it is closer to the positive one, then its score will be zero, same as any other point within the same zone. But if it is on the negative side, it will be assigned a more and more negative score the further it gets.
```
Comment (kacperlukawski) [inline qdrant-landing/content/articles/discovery-search.md:83]:
```suggestion
__Context search__ solves this by de-focusing the search around a single point. Instead, it selects points randomly from within a zone in the vector space. This search is the most influenced by _triplet loss_, as the score can be thought of as _"how much a point is closer to a negative than a positive vector?"_. If it is closer to the positive one, then its score will be zero, same as any other point within the same zone. But if it is on the negative side, it will be assigned a more and more negative score the further it gets.
```

**Review by kacperlukawski — COMMENTED** [review]
Great article overall 👏  I'm wondering if the visuals could be improved. If each line divides the space into two zones, they could have a transparent background (green or red, depending on whether it's positive or negative). Multiple splits should result in an overlap of the colors so that our preferred area is the greenest part. Does that make sense?

File: qdrant-landing/content/articles/discovery-search.md (hunk 1 above)
Comment (coszio) [reply qdrant-landing/content/articles/discovery-search.md]: Upgraded it, @agourlay, is it clearer now?

File: qdrant-landing/content/articles/discovery-search.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+This is where the concept of __vector _context___ comes in. We define _context_ as a list of pairs, which in turn are made up of a positive and a negative vector. By adding a context, we can define boundaries within the vector space, to which the positive side will always be preferred over the negative side, effectively partitioning the space in which the search is performed. After the space is partitioned, we then need a _target_ to select the points that are closest to it.
+
+![Discovery search visualization](/articles_data/discovery-search/discovery-search.png)
+
+When hearing about positive and negative vectors, one might be taken back to recommendation systems. However, now these positive and negative vectors have to be provided in the shape of a context. This is inspired from the machine-learning concept of _triplet loss_, where you have three vectors: an anchor, a positive, and a negative. But in this case, the anchor would be the vector you are searching for, and the positive and negative vectors are the ones that define the context.
```
[the remark this answers is not in the document]
Comment (coszio) [reply qdrant-landing/content/articles/discovery-search.md:34]: I've actually never used triplet-loss, would you elaborate?

File: qdrant-landing/content/articles/discovery-search.md (hunk 3 above)
Comment (coszio) [reply qdrant-landing/content/articles/discovery-search.md:61]: Nice!! I like that some people are already aware of this problem

**Review by kacperlukawski — APPROVED** [review]

File: qdrant-landing/content/articles/discovery-search.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+__Context search__ solves this by de-focusing the search around a single point. Instead, it selects points randomly from within a zone in the vector space. This search is the most influenced by _triplet loss_, as the score can be thought of as _"how much a point is closer to a negative than a positive vector?"_. If it is closer to the positive one, then its score will be zero, same as any other point within the same zone. But if it is on the negative side, it will be assigned a more and more negative score the further it gets.
+
+![Context search visualization](/articles_data/discovery-search/context-search.png)
+
+Creating complex tastes in a high-dimensional space becomes easier, since you can just add more context pairs to the search. So, in theory, you should be able to constrain the space enough so you select points from a per-search "category" created just from the context in the input.
+
+![A more complex context search](/articles_data/discovery-search/complex-context-search.png)
+
+This way you can give refeshing recommendations, while still being in control by providing positive and negative feedback.
```
Comment (elbart) [inline qdrant-landing/content/articles/discovery-search.md:89]: I found that section quite well explained - was easy to follow for me. however you really got me with the kicker in the front - with the music recommendation. having a real example would be so cool :)

File: qdrant-landing/content/articles/discovery-search.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  - specialty
+  - search
+  - discovery
+  - state of the art
+  - vector search
+---
+
+When Christopher Columbus and his crew sailed to cross the Atlantic Ocean, they were not looking for America. They were looking for a new route to India, and they were convinced that the Earth was round. They didn't know anything about America, but since they were going west, they stumbled upon it.
+
+They couldn't reach their _target_, because the the geography didn't let them, but once they realized it wasn't India, they claimed it a new "discovery" for their crown. If we consider that sailors need bodies of water to sail, then we can establish a _context_ which is positive in the water, and negative on land. Once the sailor's search was stopped by the land, they could not go any further, and a new route was found. Let's keep this concepts of _target_ and _context_ in mind as we explore the new functionality of Qdrant: __Discovery search__.
```
[the remark this answers is not in the document]
Comment (elbart) [reply qdrant-landing/content/articles/discovery-search.md:24]: I like the analogy here and it really helps to to get into the topic. I'd just remove this `bodies of`phrase - confuses me. it's just about water and land, right?

**Review by elbart — COMMENTED** [review]
All in all, a great article already. I really understood most of it just by carefully reading.

File: qdrant-landing/content/articles/discovery-search.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+title: "Discovery needs context" #required
+short_description: Discover points by constraining the space
+description: Qdrant released a new functionality that lets you constrain the space in which a search is performed, relying only pure vectors #required
+social_preview_image: /articles_data/discovery-search/social_preview.jpg # This image will be used in social media previews, should be 1200x630px. Required.
+small_preview_image: /articles_data/discovery-search/icon.svg # This image will be used in the list of articles at the footer, should be 40x40px
+preview_dir: /articles_data/discovery-search/preview # This directory contains images that will be used in the article preview. They can be generated from one image. Read more below. Required.
+weight: -100 # This is the order of the article in the list of articles at the footer. The lower the number, the higher the article will be in the list.
+author: Luis Cossío # Author of the article. Required.
+author_link: https://coszio.github.io # Link to the author's page. Required.
+date: 2023-12-11T13:00:00+03:00 # Date of the article. Required.
```
Comment (mjang) [inline qdrant-landing/content/articles/discovery-search.md:11]: I presume you'll want to change the date

**Review by mjang — COMMENTED** [review]
@coszio I love your style! It's easy to read. I've suggested some light edits, as I want to preserve your voice.

I'm now starting my 4th week at Qdrant. I'm thinking we should have different standards for our documentation, articles, and blogs. FYI, I'm working on partial automation using the tools at https://vale.sh. If you're interested, I'm working through some customizations in a `vale_test` branch.

If this were a change to our `documentation/`, I'd enforce some additional standards based on the Google Developer Style Guide.

File: qdrant-landing/content/articles/discovery-search.md (hunk 8 above)
Comment (coszio) [reply qdrant-landing/content/articles/discovery-search.md:11]: Yes, as soon as I know the publication date 😄

**Review by mjang — APPROVED** [review]
Thanks. I have no additional comments.
